### PR TITLE
READY: Add missing dependencies

### LINKED ISSUE #5027: Missing dependencies in dependencies.py
It seems that we are missing two dependencies:
- **PIL**, required by the GUI
- **pyasn1**, required by IPv8

Both of them can easily be installed with pip3.

### PR DESCRIPTION
**Pull request by Solomon1732**
Added missing dependencies. Closes #5027

### CODE REVIEW

**Comment by Solomon1732** [comment]
```
ERROR:ExperimentRunner:Experiment execution failed, exiting with error.
ERROR:ExperimentRunner:<twisted.python.failure.Failure twisted.internet.defer.FirstError: FirstError[#1, [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ProcessTerminated'>: A process has ended with a probable error condition: process ended with exit code 1.
]]>
Unhandled Error
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/twisted/internet/base.py", line 428, in fireEvent
    DeferredList(beforeResults).addCallback(self._continueFiring)
  File "/usr/local/lib/python2.7/dist-packages/twisted/internet/defer.py", line 321, in addCallback
    callbackKeywords=kw)
  File "/usr/local/lib/python2.7/dist-packages/twisted/internet/defer.py", line 310, in addCallbacks
    self._runCallbacks()
  File "/usr/local/lib/python2.7/dist-packages/twisted/internet/defer.py", line 653, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
--- <exception caught here> ---
  File "/usr/local/lib/python2.7/dist-packages/twisted/internet/base.py", line 441, in _continueFiring
    callable(*args, **kwargs)
exceptions.SystemExit: 1
```
Odd. Why the error? All I did was added dependencies...

**Comment by Solomon1732** [comment]
Wait... `python2.7`? Isn't it suppose to be python 3.something?

File: Tribler/dependencies.py [hunk 1]
```diff
@@ -27,6 +27,8 @@
     {'module': 'netifaces', 'install_type': 'pip3', 'package': 'netifaces', 'optional': False, 'scope': 'core'},
     {'module': 'six', 'install_type': 'pip3', 'package': 'six', 'optional': False, 'scope': 'both'},
     {'module': 'bitcoinlib', 'install_type': 'pip3', 'package': 'bitcoinlib', 'optional': True, 'scope': 'core'},
+    {'module': 'PIL', 'install_type': 'pip3', 'package': 'PIL', 'optional': False, 'scope': 'gui'},
+    {'module': 'pyasn1', 'install_type': 'pip3', 'package': 'pyasn1', 'optional': False, 'scope': 'ipv8'},
```
Comment (devos50) [inline Tribler/dependencies.py]: IIRC, the scope should either be `gui` or `core`. In this case, `pyasn1` is used by the `core`.

**Comment by devos50** [comment]
@Solomon1732 We are currently working on the PR checks. @xoriole or @ichorid can you report on the status of this? Fixing the PR checkers has some priority since we cannot reliably merge open PRs anymore when checks are disabled/failing.

**Comment by ichorid** [comment]
retest this please

**Comment by ichorid** [comment]
retest this please

**Comment by ichorid** [comment]
retest this please

File: Tribler/dependencies.py (hunk 1 above)
Comment (Solomon1732) [reply Tribler/dependencies.py]: Done.

**Comment by Solomon1732** [comment]
The only exception I could see is from the \[Py3\]\[Async\] Tests on Linux:
```
ERROR:ExperimentRunner:Experiment execution failed, exiting with error.
ERROR:ExperimentRunner:<twisted.python.failure.Failure twisted.internet.defer.FirstError: FirstError[#1, [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ProcessTerminated'>: A process has ended with a probable error condition: process ended with exit code 1.
]]>
Unhandled Error
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/twisted/internet/base.py", line 428, in fireEvent
    DeferredList(beforeResults).addCallback(self._continueFiring)
  File "/usr/local/lib/python2.7/dist-packages/twisted/internet/defer.py", line 321, in addCallback
    callbackKeywords=kw)
  File "/usr/local/lib/python2.7/dist-packages/twisted/internet/defer.py", line 310, in addCallbacks
    self._runCallbacks()
  File "/usr/local/lib/python2.7/dist-packages/twisted/internet/defer.py", line 653, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
--- <exception caught here> ---
  File "/usr/local/lib/python2.7/dist-packages/twisted/internet/base.py", line 441, in _continueFiring
    callable(*args, **kwargs)
exceptions.SystemExit: 1
```
Using python2.7, which is odd considering the test is under python 3.

The other tests fail because "no tests reported for the file" for JUnit and XUnit tests. The Linux test I mentioned earlier also has the same JUnit error.

I can't seem to figure out why the tests are failing.

Also looking at the file I can see `from __future__ import absolute_import, print_function`
https://github.com/Solomon1732/tribler/blob/f_missing_dependencies/Tribler/dependencies.py#L6
which I guess isn't needed considering this is python 3?

**Comment by qstokkink** [comment]
@Solomon1732 I'm not too up to speed with the state of the Tribler repo, but I believe you can safely ignore the tester output as the Tribler testing environment is horribly broken during the ongoing full migration to Python 3.

**Comment by ichorid** [comment]
@Solomon1732 , could you please start working on `devel` branch instead? There are many more things that should be fixed, as we recently finished complete refactoring of the Core, that includes moving to Asyncio, Pathlib, and change of repository structure to conform with PEP8 and other Python industry standards. `release-7.4.0` branch is stable now, but is still on hold until Usenix experiments are finished, which will happen at January 15.

So please, move your attention to `devel` branch. For instance, you can [remove the remains of `path_util` compatibility wrapper](https://github.com/Tribler/tribler/issues/5030). Or generally hunt for dead code, possibly with some automated utils like `vulture`? You can start by reviewing utility functions in `tribler_core/utils` dir. These haven't seen any attention in *ages*.

**Comment by Solomon1732** [comment]
Sure thing 😄 I wanted to merge it to the `7.4.0` branch since it was mentioned in #5027 that this is the branch to target with the PR, but if it's too much I don't mind targeting a sister PR to devel and closing this one 😄

**Comment by ichorid** [comment]
That is correct to target it on *this* branch, since this is a bugfix, and bugfixes go onto `release-*` branch. So, do not close this PR. I just meant that the specific situation at this moment is that `release-7.4.0` is stable enough, and that is better to point your attention to `devel`, especially considering the refactoring situation.

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
All [Py2] and [Py3] tests are passing. I think this PR can be merged.

There are other failing tests
- Build documentation - Can be fixed in a separate PR
- MacOS GUI tests (Python2 compatibility issue with print) - that is ok, will fix the tester later
- [Py3][Async] Tests - these tests are for Tribler 7.5.X+ so expected to fail

**Pull request merged**